File: java/core/src/main/java/network/aika/typedefs/NodeDefinition.java
package network.aika.typedefs;

import network.aika.activations.Activation;
import network.aika.type.TypeRegistry;
import network.aika.neurons.Neuron;

public class NodeDefinition {

    public final NeuronDefinition neuron;

    public final ActivationDefinition activation;

    public NodeDefinition(TypeRegistry registry, String name) {
        neuron = new NeuronDefinition(registry, name + "Neuron");
        activation = new ActivationDefinition(registry, name + "Activation");

        neuron.setActivation(activation);
        activation.setNeuron(neuron);
    }

    public NodeDefinition addParent(NodeDefinition parent) {
        neuron.addParent(parent.neuron);
        activation.addParent(parent.activation);
        return this;
    }
}

File: src/network/node_definition.cpp
#include "network/node_definition.h"

NodeDefinition::NodeDefinition(TypeRegistry* registry, const std::string& name) : registry(registry), name(name), neuron(nullptr), activation(nullptr) {}

NodeDefinition* NodeDefinition::addParent(NodeDefinition* parent) {
    // Implementation for adding a parent node
    return this;
} 